internal/distributed/proxy/httpserver/handler_v1.go

```diff
@@ -0,0 +1,582 @@
+}
+
+func (h *Handlers) getCollectionDetails(c *gin.Context) {
+	collectionName := c.Query(HTTPCollectionName)
+	if collectionName == "" {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "collectionName is required."})
+		return
+	}
+	dbName := c.DefaultQuery(HTTPDbName, DefaultDbName)
+	coll, err := h.describeCollection(c, dbName, collectionName, true)
+	if err != nil {
+		return
+	}
+	stateResp, stateErr := h.proxy.GetLoadState(c, &milvuspb.GetLoadStateRequest{
+		CollectionName: collectionName,
+	})
+	collLoadState := ""
+	if stateErr != nil {
+		log.Warn("get collection load state fail", zap.String("collection", collectionName), zap.String("err", stateErr.Error()))
+	} else if stateResp.Status.ErrorCode != commonpb.ErrorCode_Success {
+		log.Warn("get collection load state fail", zap.String("collection", collectionName), zap.String("err", stateResp.Status.Reason))
+	} else {
+		collLoadState = stateResp.State.String()
+	}
+	vectorField := ""
+	for _, field := range coll.Schema.Fields {
+		if field.DataType == schemapb.DataType_BinaryVector || field.DataType == schemapb.DataType_FloatVector {
+			vectorField = field.Name
+			break
+		}
+	}
+	indexResp, indexErr := h.proxy.DescribeIndex(c, &milvuspb.DescribeIndexRequest{
+		CollectionName: collectionName,
+		FieldName:      vectorField,
+	})
+	var indexDesc []gin.H
+	if indexErr != nil {
+		log.Warn("get indexes description fail", zap.String("collection", collectionName), zap.String("vectorField", vectorField), zap.String("err", indexErr.Error()))
+	} else if indexResp.Status.ErrorCode != commonpb.ErrorCode_Success {
+		log.Warn("get indexes description fail", zap.String("collection", collectionName), zap.String("vectorField", vectorField), zap.String("err", indexResp.Status.Reason))
+	} else {
+		indexDesc = printIndexes(indexResp.IndexDescriptions)
+	}
+	c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusOK, HTTPReturnData: gin.H{
+		HTTPCollectionName:    coll.CollectionName,
+		HTTPReturnDescription: coll.Schema.Description,
+		"fields":              printFields(coll.Schema.Fields),
+		"indexes":             indexDesc,
+		"load":                collLoadState,
+		"shardsNum":           coll.ShardsNum,
+		"enableDynamic":       coll.Schema.EnableDynamicField,
+	}})
+}
+
+func (h *Handlers) dropCollection(c *gin.Context) {
+	httpReq := DropCollectionReq{
+		DbName: DefaultDbName,
+	}
+	if err := c.ShouldBindBodyWith(&httpReq, binding.JSON); err != nil {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "check your parameters conform to the json format", HTTPReturnError: err.Error()})
+		return
+	}
+	if httpReq.CollectionName == "" {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "collectionName is required."})
+		return
+	}
+	has, err := h.hasCollection(c, httpReq.DbName, httpReq.CollectionName)
```

Comment:
There is no necessity to ascertain the presence of the Collection; the operation of "DropCollection" is idempotent.

Reply:
To be consistent with the Cloud platform, it return error where drop a non-exist collection
https://docs.zilliz.com/reference/drop-collection#possible-errors